trieste/acquisition/sampler.py

```diff
@@ -99,8 +143,31 @@ class GumbelSampler(DiscreteSampler):
     Samples are obtained via the Gumbel distribution by sampling :math:`r` uniformly from
     :math:`[0, 1]` and applying the inverse probability integral transform
     :math:`y = \mathcal G^{-1}(r; a, b)`.
+
+
+    Note that the :class:`GumbelSampler` can only sample a function's minimal value and not
+    its minimiser.
     """
 
+    def __init__(self, sample_size: int, model: ProbabilisticModel, sample_min_value: bool = False):
```

Comment:
The init functions don't need to be the same for all the ThompsonSamplers: methods do (since you might not know which type of thompson sampler you're calling them on) but for a constructor you always know the type. Since GumbelSampler doesn't support sampling the minimiser then there's no point in including `sample_min_value` as an argument. Just call `super().__init__(sample_size, model, True)` at the end.

```suggestion
    def __init__(self, sample_size: int, model: ProbabilisticModel):
```